wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -103,8 +102,9 @@ public Response authenticate(@Required @QueryParam("oauth_provider") String oaut
 
         OAuthAuthenticator oauth = getAuthenticator(oauthProvider);
         if (!isNullOrEmpty(userId) && !userId.equals(EnvironmentContext.getCurrent().getSubject().getUserId())) {
-            throw new ForbiddenException(
-                    "Provided userId " + userId + " is not related to current user " + EnvironmentContext.getCurrent().getSubject().getUserId());
+            throw new ForbiddenException("Provided userId " + userId + " is not related to current user " + EnvironmentContext.getCurrent()
```

Comment:
`EnvironmentContext.getCurrent()` returns `null`. The method worked earlier because `userId` was always `null` and `EnvironmentContext.getCurrent()` was never called. So I've removed the check and `userId` query parameter.